added draggable property to ScrollView, closes #2260

codeshaunted opened the pull request:
Adds a `draggable` property to ScrollView that binds to the `interactive` property of the internal Flickable. Should close #2260.

<!--
- [ ] If the change modifies a visible behavior, it changes the documentation accordingly
- [ ] If possible, the change is auto-tested
- [ ] If the changes fixes or close an existing issue, the commit message reference the issue with `Fixes #xxx` or `Closes #xxx`
- [ ] If the change is noteworthy, the commit message should contain `ChangeLog: ...`
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/slint-ui/slint?pullRequest=8512) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/slint-ui/slint?pullRequest=8512) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/slint-ui/slint?pullRequest=8512) it.</sub>

ogoffart commented on internal/compiler/widgets/material/scrollview.slint:
```diff
@@ -117,6 +118,7 @@ export component ScrollView {
     flickable := Flickable {
         x: 0;
         y: 0;
+        interactive: false;
```
This defaults to false because on android we should default to draggable.

codeshaunted replied:
I'm a bit confused, wouldn't I want `interactive: true;` to default to draggable? And, does this also apply for Cupertino?

ogoffart replied:
right sorry my comment was not clear.
It currently defaults to true for the material style (as opposed to all the other style which explicitly set `interactive: false`

The material style is different because this is a style which is primarily interacted with touch because of android.
While the other style are to be used on desktop.

It is true that cupertino is currently the default on iOS, but our iOS support is not finished yet.

ogoffart reviewed (commented):
Thanks a lot for the PR!

This is a nice improvement and the code looks good.

I’m not sure if `draggable` is the best name. Maybe something like `mouse-drag` or `enable-mouse-pan` would be clearer. The reason is that in the future, we want `ScrollView` to always support dragging with a finger (touch), even if this property is false. But to do that, we first need support for detecting if the input is from a mouse or a finger (see https://github.com/slint-ui/slint/issues/4352).

Right now, this property is only about dragging with the mouse, so the name should make that clear.

Also, we should keep the default behavior in the Material style, to dragging is enabled by default on android. It would be good to mention in the docs that the default can depend on the style.

What do you think?

codeshaunted commented:
I've renamed the property to `mouse-drag` to hint that it's only for the mouse. I've also removed the setting to false for Material style, and added a note in the docs about this. I'm not sure if this is the best way to denote it in the docs, so please let me know if there's a better way. Thanks for the input!

ogoffart commented:
@FloVanGH / @tronical  any opinion on the name of the property (`mouse-drag`)?

ogoffart commented:
Thanks a lot!
